edx_exams/apps/api/v1/views.py

```diff
@@ -1 +1,287 @@
+
+    # TODO: The resource link should uniquely represent the assessment in the Assessment Platform.
+    # TODO: We SHOULD provide a value for the title attribute.
+    # TODO: It's RECOMMENDED to provide a value for the description attribute.
+    # TODO: The xblock-lti-consumer library does not currently support setting these attributes.
+    resource_link = get_resource_link()
+
+    lti_consumer.enable_proctoring(
+        # NOTE TO SELF: attempt_number is an auto-incrementing integer from 1 per learner, per assessment.
+        29, # attempt_number
+        session_data=session_data,
+        start_assessment_url=start_assessment_url,
+    )
+
+    # This is necessary for testing with the IMS tool, since the user will be an AnonymousUser without an id.
+    # TODO: Replace this with the authenticated user's id.
+    # TODO: Remove this once testing is complete.
+    user_id = 1 if request.user.id is None else request.user.id
+
+    # Required user claim data
+    lti_consumer.set_user_data(
+        #   user_id=request.user.id,
+        user_id=user_id,
+        # Pass Django user role to library
+        role='student'
+    )
+
+    # These claims are optional.
+    # TODO: This will need to have additional consideration for PII.
+    optional_user_identity_claims = get_optional_user_identity_claims()
+    lti_consumer.set_proctoring_user_data(
+        **optional_user_identity_claims
+    )
+
+    context = {}
+
+    # Authorization Servers MUST support the use of the HTTP GET and POST methods defined in RFC 2616 [RFC2616]
+    # at the Authorization Endpoint.
+    # See 3.1.2.1.Authentication Request of the OIDC specification.
+    # https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest
+    preflight_response_method = request.method
+    preflight_response = request.GET if preflight_response_method == 'GET' else request.POST
+
+    context.update({
+    'preflight_response': preflight_response.dict(),
+    'launch_request': lti_consumer.generate_launch_request(
+        preflight_response,
+        resource_link,
+    )
+    })
+
+    # This template renders an auto-submitting form, which makes a POST request to the redirect_uri, specified in the Tool's response
+    # to the request the Assessment Platform made to the Tool's OIDC login initiation URL.
+    return render(request, 'api/lti_launch_request_form.html', context)
+
+# We do not want Django's CSRF protection enabled for POSTs made by external services to this endpoint.
+# Please see the comment for the lti_oidc_authentication view for a more detailed justification.
+@csrf_exempt
+# Per the Proctoring Services specification, the Tool should only make a POST request to this endpoint.
+@require_http_methods(['POST'])
+def lti_start_assessment(request):
+    """
+    This is the view that the Tool will make a request to when the leaner is ready to begin the assessment
+    from the Tool's point of view.
```

Comment:
it might be helpful to have a comment at the top of the file with the standard call ordering we expect to see for an exam. And possibly split the view file into views called from edx vs from redirect by the tool OR calls which start vs calls which end the exam. Which is two different possible splits.  Mostly this just keeps the amount to digest in a single file down.